crates/astria-sequencer/src/app.rs

```diff
@@ -334,17 +346,24 @@ impl App {
     ///
     /// Note that the first two "transactions" in the block, which are the proposer-generated
     /// commitments, are ignored.
-    #[instrument(name = "App::deliver_tx_after_execution", skip(self))]
-    pub(crate) fn deliver_tx_after_execution(
+    #[instrument(name = "App::deliver_tx_after_proposal", skip(self))]
+    pub(crate) async fn deliver_tx_after_proposal(
         &mut self,
-        tx_hash: &[u8; 32],
+        tx: abci::request::DeliverTx,
     ) -> Option<anyhow::Result<Vec<abci::Event>>> {
         if self.processed_txs < 2 {
             self.processed_txs += 1;
             return Some(Ok(vec![]));
         }
 
-        self.execution_result.remove(tx_hash)
+        // Process and prepare proposal are not called for full nodes, so we must
+        // exeucte the tx.
+        if !self.is_validator {
+            self.execute_block_data(vec![tx.tx.clone()]).await;
```

Comment:
just call `self.deliver_tx()` here (after decoding into `SignedTransaction`)
```suggestion
            return self.deliver_tx(signed_transaction]).await
```